|feature|operation|name|input|output_accepted|output_preferred|data|tags|description|varnomen_reference|test_source|gene|rsid|pytest_outcome|
|--- |--- |--- |--- |--- |--- |--- |--- |--- |--- |--- |--- |--- |--- |
|parsing|parse|g_del|NC_000020.10:g.278701_278703delGGC|True|True|GRCh37|RefSeq|hgvs_g|del|Tests whether the tool deconstructs the HGVS expression into its individual parts|http://varnomen.hgvs.org/recommendations/DNA/variant/deletion/|https://www.ncbi.nlm.nih.gov/projects/SNP/snp_ref.cgi?rs=6147263|ZCCHC3|rs6147263|failed|
|parsing|parse|g_del|NC_000020.11:g.298057_298059delGGC|True|True|GRCh38|RefSeq|hgvs_g|del|Tests whether the tool deconstructs the HGVS expression into its individual parts|http://varnomen.hgvs.org/recommendations/DNA/variant/deletion/|https://www.ncbi.nlm.nih.gov/projects/SNP/snp_ref.cgi?rs=6147263|ZCCHC3|rs6147263|failed|
|projection|project_g_to_t|g_to_c|NC_000020.10:g.278701_278703delGGC|NM_033089.6:c.471_473delGGC|NM_033089.6:c.471_473del|NM_033089.6:c.471_473del|GRCh37|RefSeq|hgvs_g|hgvs_c|del|Tests whether the tool projects a genomic variant to a coding transcript variant|http://varnomen.hgvs.org/recommendations/DNA/variant/deletion/|https://www.ncbi.nlm.nih.gov/projects/SNP/snp_ref.cgi?rs=6147263|ZCCHC3|rs6147263|failed|
|projection|project_g_to_t|g_to_c|NC_000020.11:g.298057_298059delGGC|NM_033089.6:c.471_473delGGC|NM_033089.6:c.471_473del|NM_033089.6:c.471_473delGGC|GRCh38|RefSeq|hgvs_g|hgvs_c|del|Tests whether the tool projects a genomic variant to a coding transcript variant|http://varnomen.hgvs.org/recommendations/DNA/variant/deletion/|https://www.ncbi.nlm.nih.gov/projects/SNP/snp_ref.cgi?rs=6147263|ZCCHC3|rs6147263|failed|
|projection|project_c_to_p|c_to_p|NM_033089.6:c.471_473delGGC|NP_149080.2:p.(Ala158del)|NP_149080.2:p.Ala158del|NP_149080.2:p.A158del|NP_149080.2:p.(A158del)|NP_149080.2:p.Ala158del|RefSeq|hgvs_c|hgvs_p|del|inframe_deletion|Tests whether the tool projects a coding transcript variant to a protein variant|http://varnomen.hgvs.org/recommendations/protein/variant/deletion/|https://www.ncbi.nlm.nih.gov/projects/SNP/snp_ref.cgi?rs=6147263|ZCCHC3|rs6147263|failed|
|normalization|rewrite|g_del_5'_to_3'_shift|NC_000020.10:g.278689_278691delGGC|NC_000020.10:g.278701_278703delGGC|NC_000020.10:g.278701_278703del|NC_000020.10:g.278701_278703del|GRCh37|hgvs_g|del|Tests whether a tool reformats a variant that is 5' shifted to 3' shifted.|http://varnomen.hgvs.org/recommendations/DNA/variant/deletion/|https://www.ncbi.nlm.nih.gov/projects/SNP/snp_ref.cgi?rs=6147263|ZCCHC3|rs6147263|failed|
|normalization|rewrite|g_del_5'_to_3'_shift|NC_000020.11:g.298048_298050delGGC|NC_000020.11:g.298057_298059delGGC|NC_000020.11:g.298057_298059del|NC_000020.11:g.298057_298059del|GRCh38|hgvs_g|del|Tests whether a tool reformats a variant that is 5' shifted to 3' shifted.|http://varnomen.hgvs.org/recommendations/DNA/variant/deletion/|https://www.ncbi.nlm.nih.gov/projects/SNP/snp_ref.cgi?rs=6147263|ZCCHC3|rs6147263|failed|
|validation|validate|validate_correct_g_del|NC_000020.10:g.278701_278703del|True|True|GRCh37|RefSeq|hgvs_g|del|Tests whether the tool validates a correctly formatted variant|http://varnomen.hgvs.org/recommendations/DNA/variant/deletion/|https://www.ncbi.nlm.nih.gov/projects/SNP/snp_ref.cgi?rs=6147263|ZCCHC3|rs6147263|failed|
|parsing|parse|g_snv|NC_000023.10:g.73501562T>C|True|True|GRCh37|RefSeq|hgvs_g|snv|Tests whether the tool deconstructs the HGVS expression into its individual parts|http://varnomen.hgvs.org/recommendations/DNA/variant/substitution/|https://www.ncbi.nlm.nih.gov/projects/SNP/snp_ref.cgi?rs=174151|FTX|rs174151|failed|
|parsing|parse|g_snv|NC_000023.11:g.74281727T>C|True|True|GRCh38|RefSeq|hgvs_g|snv|Tests whether the tool deconstructs the HGVS expression into its individual parts|http://varnomen.hgvs.org/recommendations/DNA/variant/substitution/|https://www.ncbi.nlm.nih.gov/projects/SNP/snp_ref.cgi?rs=174151|FTX|rs174151|failed|
|parsing|parse|n_snv|NR_028379.1:n.345A>G|True|True|RefSeq|hgvs_n|snv|Tests whether the tool deconstructs the HGVS expression into its individual parts|http://varnomen.hgvs.org/recommendations/DNA/variant/substitution/|https://www.ncbi.nlm.nih.gov/projects/SNP/snp_ref.cgi?rs=174151|FTX|rs174151|failed|
|projection|project_g_to_t|g_to_n|NC_000023.10:g.73501562T>C|NR_028379.1:n.345A>G|NR_028379.1:n.345A>G|GRCh37|RefSeq|hgvs_g|hgvs_n|snv|Tests whether the tool projects a genomic variant to a non-coding DNA variant|http://varnomen.hgvs.org/recommendations/DNA/variant/substitution/|https://www.ncbi.nlm.nih.gov/projects/SNP/snp_ref.cgi?rs=174151|FTX|rs174151|passed|
|projection|project_g_to_t|g_to_n|NC_000023.11:g.74281727T>C|NR_028379.1:n.345A>G|NR_028379.1:n.345A>G|GRCh38|RefSeq|hgvs_g|hgvs_n|snv|Tests whether the tool projects a genomic variant to a non-coding DNA variant|http://varnomen.hgvs.org/recommendations/DNA/variant/substitution/|https://www.ncbi.nlm.nih.gov/projects/SNP/snp_ref.cgi?rs=174151|FTX|rs174151|failed|
|projection|project_c_to_p|c_to_p_matching|NM_001135021.1:c.794T>C|NP_001128493.1:p.(Leu265Ser)|NP_001128493.1:p.(L265S)|NP_001128493.1:p.L265S|NP_001128493.1:p.Leu265Ser|NP_001128493.1:p.(Leu265Ser)|RefSeq|hgvs_c|hgvs_p|snv|missense_variant|Tests whether the tool projects a coding transcript variant to the correct protein variant when there are multiple protein accessions.|http://varnomen.hgvs.org/recommendations/DNA/variant/substitution/|https://www.ncbi.nlm.nih.gov/clinvar/variation/65464/|ELMOD3|rs587777040|failed|
|projection|project_c_to_p|c_to_p_matching|NM_001135023.1:c.794T>C|NP_001128495.1:p.(Leu265Ser)|NP_001128495.1:p.(L265S)|NP_001128495.1:p.L265S|NP_001128495.1:p.Leu265Ser|NP_001128495.1:p.(Leu265Ser)|RefSeq|hgvs_c|hgvs_p|snv|missense_variant|Tests whether the tool projects a coding transcript variant to the correct protein variant when there are multiple protein accessions.|http://varnomen.hgvs.org/recommendations/DNA/variant/substitution/|https://www.ncbi.nlm.nih.gov/clinvar/variation/65464/|ELMOD3|rs587777040|failed|
|projection|project_g_to_t|g_to_c|NC_000006.11:g.49917115C>A|NM_001166478.1:c.43G>T|NM_001166478.1:c.43G>T|GRCh37|RefSeq|hgvs_g|hgvs_c|snv|Tests whether the tool correctly projects a genomic variant to a minus strand coding transcript.|http://varnomen.hgvs.org/recommendations/DNA/variant/substitution/|https://www.ncbi.nlm.nih.gov/projects/SNP/snp_ref.cgi?rs=78790513|DEFB133|rs78790513|passed|
|projection|project_g_to_t|g_to_c|NC_000006.12:g.49949402C>A|NM_001166478.1:c.43G>T|NM_001166478.1:c.43G>T|GRCh38|RefSeq|hgvs_g|hgvs_c|snv|Tests whether the tool correctly projects a genomic variant to a minus strand coding transcript.|http://varnomen.hgvs.org/recommendations/DNA/variant/substitution/|https://www.ncbi.nlm.nih.gov/projects/SNP/snp_ref.cgi?rs=78790513|DEFB133|rs78790513|failed|
|projection|project_c_to_p|c_to_p|NM_001166478.1:c.43G>T|NP_001159950.1:p.Val15Phe|NP_001159950.1:p.V15F|NP_001159950.1:p.(Val15Phe)|NP_001159950.1:p.Val15Phe|RefSeq|hgvs_c|hgvs_p|snv|missense_variant|Tests whether the tool projects a coding transcript variant to a protein variant|http://varnomen.hgvs.org/recommendations/protein/variant/substitution/|https://www.ncbi.nlm.nih.gov/projects/SNP/snp_ref.cgi?rs=78790513|DEFB133|rs78790513|failed|
|normalization|rewrite|c_dup_shift|NM_001166478.1:c.31dup|NM_001166478.1:c.35dup|NM_001166478.1:c.35dupT|NM_001166478.1(DEFB133_v001):c.35dup|NM_001166478.1:c.35dup|RefSeq|hgvs_c|dup|3_prime_rule|Tests whether the tool correctly shifts the location to the 3 prime most shifted output|http://varnomen.hgvs.org/recommendations/DNA/variant/duplication/|https://github.com/biocommons/hgvs/blob/master/tests/test_hgvs_normalizer.py|DEFB133||passed|
|normalization|rewrite|c_ins_to_dup|NM_001166478.1:c.35_36insT|NM_001166478.1:c.35dupT|NM_001166478.1:c.35dup|NM_001166478.1(DEFB133_v001):c.35dup|NM_001166478.1:c.35dup|RefSeq|hgvs_c|dup|Tests whether the tool correctly changes an insertion to a duplication.|http://varnomen.hgvs.org/recommendations/general/|https://github.com/biocommons/hgvs/blob/master/tests/test_hgvs_normalizer.py|DEFB133||passed|
|projection|project_c_to_p|c_to_p|NM_001166478.1:c.35dup|NP_001159950.1:p.(Phe13LeufsTer18)|NP_001159950.1:p.(Phe13Leufs*18)|NP_001159950.1:p.(Phe13LeufsTer18)|RefSeq|hgvs_c|hgvs_p|dup|frameshift||http://varnomen.hgvs.org/recommendations/protein/variant/frameshift/|https://github.com/biocommons/hgvs/blob/master/tests/test_hgvs_normalizer.py|DEFB133||failed|
|normalization|rewrite|c_del_single_base|NM_001166478.1:c.31del|NM_001166478.1:c.35delT|NM_001166478.1:c.35del|NM_001166478.1(DEFB133_v001):c.35del|NM_001166478.1:c.35del|RefSeq|hgvs_c|del|Given incorrectly shifted input; parse, normalize and give the correctly 3 prime shifted output|http://varnomen.hgvs.org/recommendations/general/|https://github.com/biocommons/hgvs/blob/master/tests/test_hgvs_normalizer.py|DEFB133||passed|
|projection|project_g_to_t|g_to_c_delins|NC_000001.10:g.17345192_17345217delinsTTGGGGCAAGTAAAGGAACAGGTTC|NM_003000.2:c.*159_*184delinsGAACCTGTTCCTTTACTTGCCCCAA|NM_003000.2:c.*159_*184delinsGAACCTGTTCCTTTACTTGCCCCAA|GRCh37|RefSeq|hgvs_g|hgvs_c|delins|Tests whether the tool correctly project a genomic indel to a coding transcript|http://varnomen.hgvs.org/recommendations/DNA/variant/delins/|https://www.ncbi.nlm.nih.gov/clinvar/variation/165178/|SDHB|rs727503414|passed|
|projection|project_g_to_t|g_to_c_delins|NC_000001.11:g.17018697_17018722delinsTTGGGGCAAGTAAAGGAACAGGTTC|NM_003000.2:c.*159_*184delinsGAACCTGTTCCTTTACTTGCCCCAA|NM_003000.2:c.*159_*184delinsGAACCTGTTCCTTTACTTGCCCCAA|GRCh38|RefSeq|hgvs_g|hgvs_c|delins|Tests whether the tool correctly project a genomic indel to a coding transcript|http://varnomen.hgvs.org/recommendations/DNA/variant/delins/|https://www.ncbi.nlm.nih.gov/clinvar/variation/165178/|SDHB|rs727503414|failed|
|projection|project_t_to_g|c_to_g_delins|NM_003000.2:c.*159_*184delinsGAACCTGTTCCTTTACTTGCCCCAA|NC_000001.10:g.17345192_17345217delinsTTGGGGCAAGTAAAGGAACAGGTTC|NC_000001.10:g.17345192_17345217delinsTTGGGGCAAGTAAAGGAACAGGTTC|GRCh37|RefSeq|hgvs_g|hgvs_c|delins|Tests whether the tool correctly projects a transcript indel to a genomic sequence||https://www.ncbi.nlm.nih.gov/clinvar/variation/165178/|SDHB|rs727503414|passed|
|projection|project_g_to_t|g_to_c|NC_000011.10:g.112088971G>T|NM_003002.3:c.274G>T|NM_003002.3:c.274G>T|GRCh38|RefSeq|hgvs_g|hgvs_c|snv|Tests whether the tool projects a genomic SNV to a coding transcript|http://varnomen.hgvs.org/recommendations/general/|https://www.ncbi.nlm.nih.gov/clinvar/variation/6897/|SDHD|rs80338845|failed|
|normalization|rewrite|c_del_multiple|NM_000492.3:c.1520_1522delTCT|NM_000492.3:c.1521_1523delCTT|NM_000492.3(CFTR):c.1521_1523delCTT|NM_000492.3:c.1521_1523del|NM_000492.3(CFTR):c.1521_1523del|NM_000492.3(CFTR_v001):c.1521_1523del|NM_000492.3:c.1521_1523del|RefSeq|hgvs_c|del|3_prime_rule|Given incorrectly shifted input; parse, normalize and give the correctly 3 prime shifted output|http://varnomen.hgvs.org/recommendations/general/|https://www.ncbi.nlm.nih.gov/clinvar/variation/7105/|CFTR|rs113993960|passed|
|projection|project_g_to_t|g_to_c|NC_000007.13:g.117199646_117199648delCTT|NM_000492.3:c.1521_1523delCTT|NM_000492.3:c.1521_1523del|NM_000492.3:c.1521_1523del|GRCh37|RefSeq|hgvs_g|hgvs_c|del||http://varnomen.hgvs.org/recommendations/general/|https://www.ncbi.nlm.nih.gov/clinvar/variation/7105/|CFTR|rs113993960|passed|
|projection|project_g_to_t|g_to_c|NC_000007.14:g.117559592_117559594delCTT|NM_000492.3:c.1521_1523delCTT|NM_000492.3:c.1521_1523del|NM_000492.3:c.1521_1523del|GRCh38|RefSeq|hgvs_g|hgvs_c|del||http://varnomen.hgvs.org/recommendations/general/|https://www.ncbi.nlm.nih.gov/clinvar/variation/7105/|CFTR|rs113993960|failed|
|projection|project_c_to_p|c_to_p|NM_000492.3:c.1521_1523delCTT|NP_000483.3:p.Phe508del|NP_000483.3:p.(Phe508del)|NP_000483.3:p.Phe508del|RefSeq|hgvs_c|hgvs_p|del||http://varnomen.hgvs.org/recommendations/general/|https://www.ncbi.nlm.nih.gov/clinvar/variation/7105/|CFTR|rs113993960|failed|
|normalization|rewrite|g_ins|NC_000002.11:g.37480320_37480321insT|NC_000002.11:g.37480321dupT|NC_000002.11:g.37480321dup|NC_000002.11:g.37480321dup|GRCh37|hgvs_g|dup|Given incorrectly shifted input; parse, normalize and give the correctly 3 prime shifted output|http://varnomen.hgvs.org/recommendations/DNA/variant/duplication/|https://www.ncbi.nlm.nih.gov/projects/SNP/snp_ref.cgi?rs=151034144|PRKD3|rs151034144|failed|
|normalization|rewrite|g_ins|NC_000002.12:g.37253177_37253178insT|NC_000002.12:g.37253178dup|NC_000002.12:g.37253178dupT|NC_000002.12:g.37253178dup|GRCh38|hgvs_g|dup|Given incorrectly shifted input; parse, normalize and give the correctly 3 prime shifted output|http://varnomen.hgvs.org/recommendations/DNA/variant/duplication/|https://www.ncbi.nlm.nih.gov/projects/SNP/snp_ref.cgi?rs=151034144|PRKD3|rs151034144|failed|
|projection|project_g_to_t|g_to_c_dup|NC_000002.11:g.37480321dup|NM_005813.4:c.2673dup|NM_005813.4:c.2673dupA|NM_005813.4:c.2673dup|GRCh37|RefSeq|hgvs_g|dup|Tests whether the tool projects a genomic variant to a coding transcript variant|http://varnomen.hgvs.org/recommendations/DNA/variant/duplication/|https://www.ncbi.nlm.nih.gov/projects/SNP/snp_ref.cgi?rs=151034144|PRKD3|rs151034144|failed|
|projection|project_g_to_t|g_to_c_dup|NC_000002.12:g.37253178dup|NM_005813.4:c.2673dup|NM_005813.4:c.2673dupA|NM_005813.4:c.2673dup|GRCh38|RefSeq|hgvs_g|dup|Tests whether the tool projects a genomic variant to a coding transcript variant|http://varnomen.hgvs.org/recommendations/DNA/variant/duplication/|https://www.ncbi.nlm.nih.gov/projects/SNP/snp_ref.cgi?rs=151034144|PRKD3|rs151034144|failed|
|formatting|rewrite|c_ins_to_dup|NM_005813.3:c.2672_2673insA|NM_005813.3:c.2673dupA|NM_005813.3:c.2673dup|NM_005813.3(PRKD3_v001):c.2673dup|NM_005813.3:c.2673dup|RefSeq|hgvs_c|dup|Tests whether the tool rewrites a coding transcript duplication annotated as an insertion|http://varnomen.hgvs.org/recommendations/DNA/variant/duplication/|https://www.ncbi.nlm.nih.gov/projects/SNP/snp_ref.cgi?rs=151034144|PRKD3|rs151034144|passed|
|validation|validate|validate_correct_c_snv|NM_001197320.1:c.281C>T|True|True|Refseq|hgvs_c|snv|Tests whether the tool validates a correctly formatted variant|http://varnomen.hgvs.org/recommendations/DNA/variant/substitution/|https://www.ncbi.nlm.nih.gov/projects/SNP/snp_ref.cgi?rs=201430561|MCL1|rs201430561|failed|
|validation|validate|invalidate_incorrect_c_snv|NM_001197320.1:c.281A>T|False|False|Refseq|hgvs_c|snv|Tests whether the tool invalidates a variant where the variant reference does not agree with the reference sequence|http://varnomen.hgvs.org/recommendations/DNA/variant/substitution/||||passed|
|formatting|rewrite|allow_3_letter_aa|NP_689699.2:p.(G553E)|NP_689699.2:p.(G553E)|NP_689699.2:p.(Gly553Glu)|NP_689699.2:p.G553E|NP_689699.2:p.Gly553Glu|NP_689699.2:p.(Gly553Glu)|RefSeq|hgvs_p|snv|missense_variant|Tests whether the tool rewrites a 1 letter protein variant with 3 letter amino acid codes||https://www.ncbi.nlm.nih.gov/projects/SNP/snp_ref.cgi?rs=151238352|SAMD11|rs151238352|failed|
|validation|validate|c_repeated_sequences|NM_002111.6:c.52CAG(36_39)|False|False|RefSeq|hgvs_c|repeated_sequences|Tests whether the tool validates a correctly formatted variant|http://varnomen.hgvs.org/recommendations/DNA/variant/repeated/|https://www.ncbi.nlm.nih.gov/clinvar/RCV000032099/|HTT||passed|
|normalization|rewrite|c_repeated_sequences_3'_shift|NM_002111.6:c.52CAG(36_39)|NM_002111.6:c.53AGC(36_39)|NM_002111.6:c.53AGC(36_39)|RefSeq|hgvs_c|repeated_sequences|Tests whether the tool correctly shifts the location to the 3 prime most shifted output|http://varnomen.hgvs.org/recommendations/DNA/variant/repeated/|https://www.ncbi.nlm.nih.gov/clinvar/RCV000032099/|HTT||failed|
|parsing|parse|n_dup|NR_003051.3:n.-14_3dup17|NR_003051.3:n.-14_3dup|NR_003051.3:n.-14_3dup|RefSeq|hgvs_n|dup|Tests whether the tool parses a non coding duplication|http://varnomen.hgvs.org/recommendations/DNA/variant/duplication/;http://varnomen.hgvs.org/bg-material/numbering/#DNAc|https://www.ncbi.nlm.nih.gov/clinvar/variation/14221/|RMRP|rs878853178|failed|
|normalization|rewrite|g_ins_shift|NC_000001.10:g.78024346_78024347insATTTA|NC_000001.10:g.78024350_78024351insAATTT|NC_000001.10:g.78024350_78024351insAATTT|GRCh37|hgvs_g|ins|3_prime_rule|Tests whether the tool correctly shifts the location to the 3 prime most shifted output|http://varnomen.hgvs.org/recommendations/DNA/variant/insertion/|https://www.ncbi.nlm.nih.gov/projects/SNP/snp_ref.cgi?rs=71756472|AK5|rs71756472|failed|
|normalization|rewrite|g_ins_shift|NC_000001.11:g.77558661_77558662insATTTA|NC_000001.11:g.77558665_77558666insAATTT|NC_000001.11:g.77558665_77558666insAATTT|GRCh38|hgvs_g|ins|3_prime_rule|Tests whether the tool correctly shifts the location to the 3 prime most shifted output|http://varnomen.hgvs.org/recommendations/DNA/variant/insertion/|https://www.ncbi.nlm.nih.gov/projects/SNP/snp_ref.cgi?rs=71756472|AK5|rs71756472|failed|
|projection|project_g_to_t|g_to_c_ins|NC_000001.10:g.78024350_78024351insAATTT|NM_012093.3:c.1606_1607insAATTT|NM_012093.3:c.1606_1607insAATTT|GRCh37|RefSeq|hgvs_g|hgvs_c|ins|3_prime_rule|Tests whether the tool correctly projects a genomic variant to a coding transcript.|http://varnomen.hgvs.org/recommendations/DNA/variant/insertion/|https://www.ncbi.nlm.nih.gov/projects/SNP/snp_ref.cgi?rs=71756472|AK5|rs71756472|passed|
|normalization|rewrite|c_ins_shift|NM_012093.3:c.1602_1603insATTTA|NM_012093.3:c.1606_1607insAATTT|NM_012093.3(AK5_v001):c.1606_1607insAATTT|NM_012093.3:c.1606_1607insAATTT|RefSeq|hgvs_c|ins|3_prime_rule|Tests whether the tool correctly shifts the location to the 3 prime most shifted output|http://varnomen.hgvs.org/recommendations/DNA/variant/insertion/|https://www.ncbi.nlm.nih.gov/projects/SNP/snp_ref.cgi?rs=71756472|AK5|rs71756472|passed|
|projection|project_c_to_p|c_to_p|NM_012093.3:c.1606_1607insAATTT|NP_036225.2:p.(Phe536Ter)|NP_036225.2:p.Phe536Ter|NP_036225.2:p.(Phe536Ter)|RefSeq|hgvs_c|hgvs_p|ins|nonsense|Tests whether the tool correctly projects a coding variant to a protein transcript.|http://varnomen.hgvs.org/recommendations/protein/variant/substitution/|https://www.ncbi.nlm.nih.gov/projects/SNP/snp_ref.cgi?rs=71756472|AK5|rs71756472|failed|
	
